File: app/src/main/java/de/bornholdtlee/defaultproject/ui/BaseFragment.java
package de.bornholdtlee.defaultproject.ui;

import android.support.v4.app.Fragment;

import de.bornholdtlee.defaultproject.injection.components.DefaultApplicationComponent;

public abstract class BaseFragment extends Fragment {

    public DefaultApplicationComponent getDefaultApplicationComponent() {
        return ((BaseActivity) getActivity()).getDefaultApplicationComponent();
    }
}

File: app/src/main/java/de/bornholdtlee/defaultproject/ui/BaseFragment.kt
package de.bornholdtlee.defaultproject.ui

import android.support.v4.app.Fragment

import de.bornholdtlee.defaultproject.injection.components.DefaultApplicationComponent

abstract class BaseFragment : Fragment() {

    val defaultApplicationComponent: DefaultApplicationComponent
        get() = (activity as BaseActivity).defaultApplicationComponent
}